### PR TITLE
Update Python requirement docs for 2.16

### LINKED ISSUE #80428: Drop support for the oldest Python on the controller
### Summary

Drop support for the oldest Python on the controller.

### Issue Type

Feature Idea

### Component Name

`ansible-test`

### LINKED ISSUE #80524: Drop support for Python 3.5
### Summary

Drop support Python 3.5.

### Issue Type

Feature Idea

### Component Name

`ansible-test`

### PR DESCRIPTION
**Pull request by s-hertel**
##### SUMMARY
Updated docs and added the supported Python versions for 2.16 ~to the release and maintenance page~

Updating the  the release and maintenance page in https://github.com/ansible/ansible/pull/80974

2.16 versions enforced in:
* #80972
* #80973

##### ISSUE TYPE
- Docs Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by jborean93 — APPROVED** [review]

**Review by mattclay — CHANGES_REQUESTED** [review]
This should be split into two separate PRs, one for 3.5 and another for 3.9. That will allow merging the changes for 3.5 first, while the 3.9 changes remain blocked on other issues.

Both removals should be [announced](https://github.com/ansible-collections/news-for-maintainers/issues) ahead of time so collection maintainers have some advance notice.

File: docs/docsite/rst/reference_appendices/release_and_maintenance.rst
Comment (mattclay) [inline docs/docsite/rst/reference_appendices/release_and_maintenance.rst]: These changes should be in a separate PR, since they're related to the 2.16 release and not the Python version removal. It's likely we'll want to backport this change.

Comment (s-hertel) [reply docs/docsite/rst/reference_appendices/release_and_maintenance.rst]: older branches are redirected to devel's release and maintenance page (fixing https://github.com/ansible/ansible/issues/71468), so I don't think this will need to be backported, @samccann does that sound right to you?

Comment (samccann) [reply docs/docsite/rst/reference_appendices/release_and_maintenance.rst]: Starting in 2.13, I began backporting all relevant docs PRS so the docs no longer point people to devel for the release/maint page etc. People really didn't like being pointed to devel. So I keep the current release (2.15 now) in sync with devel. Well, I have a ton of backports to do, but within a week or so I should have that all caught up.  So yes, separating out a 2.16 docs change would be preferable here.

Also, has there been an announcement to the community about these 2.16 python changes? Something in the bullhorn and/or ..erm.. maintainer news?

Comment (s-hertel) [reply docs/docsite/rst/reference_appendices/release_and_maintenance.rst]: @samccann Not yet, announcement will be made once it's approved.

**Review by mattclay — APPROVED** [review]
These changes look fine, as long as we're OK with them only being correct for the `devel` branch (and eventually 2.16).

**Review by samccann — APPROVED** [review]

**Comment by samccann** [comment]
@s-hertel - is this okay to merge or is it waiting on some other PR to merge first?

**Review by mattclay — CHANGES_REQUESTED** [review]
We've started the process of moving `docs` to https://github.com/ansible/ansible-documentation, so I'm marking this as "request changes" to cancel the previous approval.

**Comment by samccann** [comment]
See https://github.com/ansible-community/community-topics/issues/243 for details

**Comment by s-hertel** [comment]
Just closing so this, as it will need to be fixed in the new repo

**Pull request closed without merging**
